apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js

```diff
@@ -1,28 +1,32 @@
+// @ts-check
+const asyncHandler = require('express-async-handler');
 const path = require('path');
 const express = require('express');
 const router = express.Router();
 
 const error = require('@prairielearn/error');
 const chunks = require('../../lib/chunks');
-const ERR = require('async-stacktrace');
+const { getQuestionCourse } = require('../../lib/question');
 
-router.get('/*', function (req, res, next) {
-  const filename = req.params[0];
-  if (!filename) {
-    return next(
-      error.make(400, 'No filename provided within clientFilesQuestion directory', {
+router.get(
+  '/*',
+  asyncHandler(async function (req, res) {
+    const filename = req.params[0];
+    if (!filename) {
+      throw error.make(400, 'No filename provided within clientFilesQuestion directory', {
         locals: res.locals,
         body: req.body,
-      }),
-    );
-  }
-  const coursePath = chunks.getRuntimeDirectoryForCourse(res.locals.course);
-  const chunk = {
-    type: 'question',
-    questionId: res.locals.question.id,
-  };
-  chunks.ensureChunksForCourse(res.locals.course.id, chunk, (err) => {
-    if (ERR(err, next)) return;
+      });
+    }
+    const course = await getQuestionCourse(res.locals.question, res.locals.course);
+    const coursePath = chunks.getRuntimeDirectoryForCourse(course);
+
+    /** @type {chunks.QuestionChunk} */
+    const chunk = {
+      type: 'question',
+      questionId: res.locals.question.id,
+    };
+    await chunks.ensureChunksForCourseAsync(res.locals.course.id, chunk);
```

Comment:
I believe we might actually want this?

```suggestion
    await chunks.ensureChunksForCourseAsync(course.id, chunk);
```

Reply:
ya, I think that's right. I've fixed it.